src/arch/arm/fastmodel/iris/thread_context.hh

```diff
@@ -124,9 +120,21 @@ class ThreadContext : public gem5::ThreadContext
 
         BpInfo(Addr _pc) : pc(_pc), events(new EventList) {}
 
-        bool empty() const { return events->empty(); }
-        bool validIds() const { return !ids.empty(); }
-        void clearIds() { ids.clear(); }
+        bool
+        empty() const
+        {
+            return events->empty();
+        }
+        bool
```

Comment:
So https://clang.llvm.org/docs/ClangFormatStyleOptions.html#separatedefinitionblocks could be a solution. The downside is that it will always enforce it if enabled with SDS_Always.
However in gem5 we usually have this unofficial coding style of having a newline in general between function definitions, but not if they are short (on a single line).

So I usually see this:

```
// Notice there is no newline
int func1() { return 1; }
int funct2() { return 2; }
```

and

```
// Notice the newline
int
func3()
{
    return 3;
}

int
func4()
{
    return 4
}
```

So I would keep it as it is

Reply:
So you're suggesting that we set this to SDS_Leave?

Reply:
I think @odanrc  makes a good point. I used the SDS_Always and it looks like the price we pay for it it is relatively ok (the PR adds more lines but also avoids lines removal) considering it will make it more likely to avoid merge conflicts which is ultimately the major concern with the introduction of clang-format